src/app/services/courses.service.ts

```diff
@@ -1,94 +1,129 @@
-        + ' and in course schedules that contain descriptions for all courses offered during a particular semester',
-      topRated: false,
-      authors: []
-    }
-  ];
-
+  constructor(private filterPipe: FilterByTitlePipe, private http: HttpClient) { }
 
-  constructor(private filterPipe: FilterByTitlePipe) { }
-
-  getList() : CoursesListItem[] {
-    return this.filterPipe.transform(this.courses, this.filterValue);
+  getList() {
+      return this.http
+        .get<ICourse[]>(
+          'http://localhost:3004/courses?start=0&count=' + this.coursesCount,  
+          this.filterValue == null ? {} :{ params : { textFragment: this.filterValue } })
+        .pipe(map((items : ICourse[]) => {
+          return items.map((item: ICourse) => {
+            return {
+              id: item.id,
+              title: item.name,
+              creationDate: item.date,
+              duration: item.length,
+              description: item.description,
+              topRated: item.isTopRated,
+              authors: item.authors
+            }
+          })
+        }));
   }
 
   createCourse(item: CoursesListItem) {
-    this.courses.push(item);
+    this.http
+      .post<ICourse>('http://localhost:3004/courses', 
+        {
+          id: item.id, 
+          name: item.title,
+          date: item.creationDate,
+          length: item.duration,
+          authors: item.authors,
+          isTopRated: item.topRated
+        })
+      .pipe(catchError(this.handleError))
+      .subscribe();
   }
 
-  getItemById(id: number) : CoursesListItem {
-    return this.courses.find(course => course.id === id);
+  getItemById(id: number) {
+    return this.http
+    .get<ICourse>(
+      'http://localhost:3004/courses/' + id)
+    .pipe(map((item : ICourse) => {
+        return {
+          id: item.id,
+          title: item.name,
+          creationDate: item.date,
+          duration: item.length,
+          description: item.description,
+          topRated: item.isTopRated,
+          authors: item.authors
+      }
+    }));
   }
 
-  updateItem(item: CoursesListItem) {
-    let index = this.courses.findIndex(course => course.id === item.id);
-    if (index !== -1) {
-      this.courses[index] = item;
-    }
+  updateItemById(id: number) {
+    // let index = this.courses.findIndex(course => course.id === item.id);
+    // if (index !== -1) {
+    //   this.courses[index] = item;
+    // }
   }
 
   removeItem(item: CoursesListItem) {
     this.removeItemByiD(item.id);
   }
 
   removeItemByiD(id: number) {
-    let index = this.courses.findIndex(course => course.id === id);
-    if (index !== -1) {
-      this.courses.splice(index, 1);
-    }
+    this.http
+      .delete('http://localhost:3004/courses/' + id)
+      .pipe(catchError(this.handleError))
+      .subscribe();
   }
 
   setFilter(filter: string) {
     this.filterValue = filter;
   }
+
+  loadMore() {
```

Comment:
not the best implementation, consumer of this service has to remember get update list after calling this method, I wish consumer could just subscribe on the course list observable, and it would trigger updates when new courses are available, or when load more method called